arch/arm/Kconfig

```diff
@@ -1011,6 +1012,7 @@ config ARCH_ARMV8M
 	select ARCH_HAVE_CPUINFO
 	select ARCH_HAVE_DEBUG
 	select ARCH_HAVE_PERF_EVENTS
+	select ARMV8M_USEBASEPRI
```

Comment:
If `basepri` is a mandatory option, should all the code related to `primask` be deleted?

Reply:
Yes, it's better to remove ARMVxM_USEBASEPRI from Kconfig and always use PRIMASK, which simplify the code and boost performance. I already saw many developers who don't deeply understand cortex-m architecture use the wrong option.

Reply:
done

Reply:
but please merge armv7-m/armv8-m into one patch, and armv6-m to a new one.